samcli/lib/build/app_builder.py

```diff
@@ -407,14 +407,18 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             "dockerfile": dockerfile,
             "tag": docker_tag,
             "buildargs": docker_build_args,
-            "decode": True,
             "platform": get_docker_platform(architecture),
             "rm": True,
         }
         if docker_build_target:
             build_args["target"] = cast(str, docker_build_target)
 
-        build_logs = self._docker_client.api.build(**build_args)
+        try:
+            (build_image, build_logs) = self._docker_client.images.build(**build_args)
+            LOG.debug("%s image is built for %s function", function_name, build_image)
```

Comment:
Does `function_name` and `build_image` need to be swapped here?

Reply:
Good catch, updated!